plugin/advisor/catalog/catalog.go

```diff
@@ -2,37 +2,151 @@ package catalog
+	ViewList      []*View
+	ExtensionList []*Extension
 }
 
-// Table is the API message for an table.
+// Table is the database table.
 type Table struct {
-	Name         string
-	DatabaseName string
+	Name string
+	// CreatedTs isn't supported for ClickHouse, SQLite.
+	CreatedTs int64
+	// UpdatedTs isn't supported for SQLite.
+	UpdatedTs int64
+	Type      string
+	// Engine isn't supported for Postgres, Snowflake, SQLite.
+	Engine string
+	// Collation isn't supported for Postgres, ClickHouse, Snowflake, SQLite.
+	Collation string
+	RowCount  int64
+	// DataSize isn't supported for SQLite.
+	DataSize int64
+	// IndexSize isn't supported for ClickHouse, Snowflake, SQLite.
+	IndexSize int64
+	// DataFree isn't supported for Postgres, ClickHouse, Snowflake, SQLite.
+	DataFree int64
+	// CreateOptions isn't supported for Postgres, ClickHouse, Snowflake, SQLite.
+	CreateOptions string
+	// Comment isn't supported for SQLite.
+	Comment    string
+	ColumnList []*Column
+	// IndexList isn't supported for ClickHouse, Snowflake.
+	IndexList []*Index
+}
+
+// Index is the database index.
+type Index struct {
+	Name string
+	// This could refer to a column or an expression.
+	Expression string
+	Position   int
+	// Type isn't supported for SQLite.
+	Type    string
+	Unique  bool
+	Primary bool
+	// Visible isn't supported for Postgres, SQLite.
+	Visible bool
+	// Comment isn't supported for SQLite.
+	Comment string
+}
+
+// Column the database table column.
+type Column struct {
+	Name     string
+	Position int
+	Default  *string
+	// Nullable isn't supported for ClickHouse.
+	Nullable bool
+	Type     string
+	// CharacterSet isn't supported for Postgres, ClickHouse, SQLite.
+	CharacterSet string
+	// Collation isn't supported for ClickHouse, SQLite.
+	Collation string
+	// Comment isn't supported for SQLite.
+	Comment string
+}
+
+// View is the database view.
+type View struct {
+	Name string
+	// CreatedTs isn't supported for ClickHouse.
+	CreatedTs  int64
+	UpdatedTs  int64
+	Definition string
+	Comment    string
+}
+
+// Extension is the database extension.
+type Extension struct {
+	Name        string
+	Version     string
+	Description string
+}
+
+// HasNoTable returns true if the current database has no table.
+func (d *Database) HasNoTable() bool {
+	for _, schema := range d.SchemaList {
+		for _, table := range schema.TableList {
+			if table != nil {
+				return false
+			}
+		}
+	}
+
+	return true
+}
+
+// IndexFind is for find index.
+type IndexFind struct {
+	SchemaName string
+	TableName  string
+	IndexName  string
 }
 
-// TableFind is the API message for find table
-type TableFind struct {
-	DatabaseName string
+// FindIndex finds the index.
+func (d *Database) FindIndex(find *IndexFind) (string, []*Index) {
+	tableName := ""
+	var indexList []*Index
+	// notMatchTable is used for PostgreSQL. In PostgreSQL, the index name is unique in a schema, not a table.
+	notMatchTable := (find.SchemaName != "" && find.TableName == "")
```

Comment:
How about put the `advisor.DBType` into the `Database` then we can use different logic for different db type?